src/cargo/util/toml/mod.rs

```diff
@@ -691,11 +691,34 @@ fn default_readme_from_package_root(package_root: &Path) -> Option<String> {
 #[tracing::instrument(skip_all)]
 fn resolve_features(
     original_features: Option<&BTreeMap<manifest::FeatureName, Vec<String>>>,
+    edition: Edition,
 ) -> CargoResult<Option<BTreeMap<manifest::FeatureName, Vec<String>>>> {
-    let Some(resolved_features) = original_features.cloned() else {
+    let Some(mut resolved_features) = original_features.cloned() else {
         return Ok(None);
     };
 
+    if Edition::Edition2024 <= edition {
```

Comment:
I understand you like Yoda style more, but forgot the reason.

Anyway, this is not a blocker.

Reply:
I actually really dislike Yoda style for equality checks.  For these checks, I prefer number line order (the lower value is always on the left, so `<` and `<=` are only ever used and not `>` or `>=`)